### PR TITLE
Move nearest node number capability from geochemistry module to MOOSE…

### LINKED ISSUE #18504: Move nearest_node_number capability from geochemistry to framework
## Reason
User in https://github.com/idaholab/moose/discussions/18497 wanted to use this capability but it s in the geochemistry module, which does not make too much sense to link when looking at reactor applications. 
I dont think this is what they should have been using, but might as well do the change.

There are a few nearest node capabilities in the framework already btw (distance, value)

## Design
git mv some files

## Impact
None right now. Possibly avoid compiling an extra module for some very niche applications, like where we d want to debug a single nodal value / examine a mesh near a single location.
Possibly not good for geochemistry module, as it moves the documentation away from the module and hides it in the framework.

### PR DESCRIPTION
**Pull request by GiudGiud**
closes #18504

Did I miss any files I should have moved?

### CODE REVIEW

**Review by WilkAndy — DISMISSED** [review]
Looks fine to me, thanks :-)

**Review by loganharbour — CHANGES_REQUESTED** [review]
Requirements need to be updated

**Comment by GiudGiud** [comment]
ah good catch!
Should be fixed now

File: test/tests/postprocessors/nearest_node_number/tests [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    type = CSVDiff
+    input = nearest_node_number_2.i
+    csvdiff = nearest_node_number_2_out.csv
+    prereq = nearest_node_number_2_2procs
+    min_parallel = 2
+    max_parallel = 2
+    min_threads = 2
+    max_threads = 2
+    requirement = "The system shall have the capability to compute the nearest node number to a given point, and will choose the smallest node number if more than one node is equidistant, when the calculation employs both threads and multiple processors."
+  [../]
```
Comment (loganharbour) [inline test/tests/postprocessors/nearest_node_number/tests:44]: All of these tests can be removed. We run with these configurations already. In addition, the reqs aren't compliant with the SQA for the framework (some duplication)

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: test/tests/postprocessors/nearest_node_number/tests [hunk 2]
```diff
@@ -0,0 +1,51 @@
+[Tests]
+  issues = '#15749'
```
Comment (loganharbour) [inline test/tests/postprocessors/nearest_node_number/tests:2]: Add issue

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by andrsd — CHANGES_REQUESTED** [review]
Could we update the `tests` file such that it uses `[name]...[]` syntax rather the old `[./name]...[../]`?

**Comment by GiudGiud** [comment]
ok I ll fix those

**Review by loganharbour — APPROVED** [review]

**Review by andrsd — APPROVED** [review]

**Pull request merged**